src/common/utils/time/time.ts

```diff
@@ -59,44 +49,44 @@ export function exclusiveToInclusiveEnd(exclusiveEnd: Date): Date {
   return new Date(exclusiveEnd.valueOf() - 1);
 }
 
-export function formatTimeRange(timeRange: TimeRange, timezone: Timezone, displayYear: DisplayYear): string {
-  const { start, end } = timeRange;
-  const startWallTime = moment.tz(start, timezone.toString());
-  const endWallTime = moment.tz(end, timezone.toString());
-  const endWallTimeInclusive = getEndWallTimeInclusive(end, timezone);
-
-  let showingYear = true;
-  let formatted: string;
-  if (startWallTime.year() !== endWallTimeInclusive.year()) {
-    formatted = [startWallTime.format(FORMAT_WITH_YEAR), endWallTimeInclusive.format(FORMAT_WITH_YEAR)].join(" - ");
-  } else {
-    showingYear = displayYear === DisplayYear.ALWAYS ||
-      (displayYear === DisplayYear.IF_DIFF && !isCurrentYear(endWallTimeInclusive.year(), timezone));
-    const fmt = showingYear ? FORMAT_WITH_YEAR : FORMAT_WITHOUT_YEAR;
-    if (startWallTime.month() !== endWallTimeInclusive.month() || startWallTime.date() !== endWallTimeInclusive.date()) {
-      formatted = [startWallTime.format(FORMAT_WITHOUT_YEAR), endWallTimeInclusive.format(fmt)].join(" - ");
-    } else {
-      formatted = startWallTime.format(fmt);
-    }
+function formatDate(start: Moment, end: Moment, showingYear: boolean): string {
+  if (start.year() !== end.year()) {
+    return [start.format(FORMAT_WITH_YEAR), end.format(FORMAT_WITH_YEAR)].join(" - ");
   }
+  const fmt = showingYear ? FORMAT_WITH_YEAR : FORMAT_WITHOUT_YEAR;
+  if (start.month() !== end.month() || start.date() !== end.date()) {
+    return [start.format(FORMAT_WITHOUT_YEAR), end.format(fmt)].join(" - ");
+  }
+  return start.format(fmt);
+}
 
-  if (startWallTime.hour() || startWallTime.minute() || endWallTime.hour() || endWallTime.minute()) {
-    formatted += (showingYear ? " " : ", ");
-
-    let startTimeStr = formatTimeOfDay(startWallTime).toLowerCase();
-    const endTimeStr = formatTimeOfDay(endWallTime).toLowerCase();
+function formatTime(start: Moment, end: Moment): string {
```

Comment:
Actually, IMO, the time range format is a bit odd.

When the range is in a single date span it seems right to present is as `Jan 22, 7:20 - 13:20`.
On the other hand when it spans multiple dates it might **probably** be more natural to present
`Jan 22, 7:20 - Jan 23, 7:20` instead of
`Jan 22 - Jun 23, 7:20 - 7:20`.

Reply:
I agree, but I think this is because often this string is truncated and presenting dates is more important than time. Question is - maybe we should check all usages and ensure that value is not truncated.

Reply:
I would pick Jan 22, 7:20 - Jan 23, 7:20 and resolve issues with truncated display eventually